setup.py

```diff
@@ -1,73 +1,102 @@
+        return 'linux.x86_64.tar.xz'
     elif sys.platform == 'darwin':
-        return 'darwin-x86_64'
+        return 'darwin.x86_64.tar.xz'
     elif sys.platform == 'win32':
-        return 'exe'
+        return 'zip'
     else:
         raise ValueError('Unsupported platform')
 
 
-def get_executable_name() -> str:
+def get_artifact_name() -> str:
     return f'shellcheck-v{SHELLCHECK_VERSION}.{get_arch()}'
 
 
 def get_download_url() -> str:
-    return f'https://storage.googleapis.com/shellcheck/{get_executable_name()}'
+    return (
+        'https://github.com/koalaman/shellcheck/releases/download/'
+        f'v{SHELLCHECK_VERSION}/{get_artifact_name()}'
+    )
 
 
 def download(url: str) -> bytes:
-    with urlopen(url) as resp, urlopen(url + '.sha512sum') as shasum_resp:
+    with urlopen(url) as resp:
         code = resp.getcode()
-        scode = shasum_resp.getcode()
-        if code != HTTPStatus.OK or scode != HTTPStatus.OK:
-            raise ValueError(f'HTTP failure. Codes: {code}, {scode}')
-
-        data = resp.read()
-        actual_shasum = sha512(data).hexdigest()
-        expected_shasum = shasum_resp.read().decode('utf8').split(' ')[0]
-
-        if expected_shasum != actual_shasum:
-            raise ValueError(
-                f'SHA512 mismatch! '
-                f'expected "{expected_shasum}" but got "{actual_shasum}"',
-            )
-
-        return data
+        if code != HTTPStatus.OK:
+            raise ValueError(f'HTTP failure. Code: {code}')
+        return resp.read()
+
+
+def extract(url: str, data: bytes):
+    result = {}
+    if re.search(r'(?i)\.tar\.', url):
+        tmpf = tempfile.NamedTemporaryFile(delete=False)
+        try:
+            with tmpf:
+                tmpf.write(data)
+            with tarfile.open(tmpf.name) as tar:
+                for member in (x for x in tar.getmembers() if x.isfile()):
+                    name = member.name.rpartition('\\')[2].rpartition('/')[2]
+                    result[name] = tar.extractfile(member).read()
+        finally:
+            os.remove(tmpf.name)
+    elif url.lower().endswith('.zip'):
+        with io.BytesIO(data) as bio, zipfile.ZipFile(bio) as zipp:
+            for info in (x for x in zipp.infolist() if not x.is_dir()):
+                name = info.filename.rpartition('\\')[2].rpartition('/')[2]
+                result[name] = zipp.read(info.filename)
+    else:
+        exe = 'shellcheck' if sys.platform != 'win32' else 'shellcheck.exe'
+        result[exe] = data
+    return result
 
 
-def save_executable(data: bytes, base_dir: str):
-    exe = 'shellcheck' if sys.platform != 'win32' else 'shellcheck.exe'
-    output_path = os.path.join(base_dir, exe)
+def save_files(files: dict, base_dir: str):
     os.makedirs(base_dir)
 
-    with open(output_path, 'wb') as fp:
-        fp.write(data)
+    for name, data in files.items():
+        match = re.search(
+            r'(?i)(?:^|[\/])(?P<basename>shellcheck)(?:.*?)(?P<ext>\.exe)?$',
```

Comment:
this is far from a simple regex, care to explain what's happening here? can we use more readable `os.path.split(...)`?

Reply:
Helps transform `path/to/shellcheck-x.x.x.exe` to `shellcheck.exe` and similarly without .exe by marking the basename without a version and handling the etension (the exe is versioned in the win32 zip). Something additional is required if you want to use os.path.split to get rid of the path part (os.path.basename would be a better fit for that).

Reply:
let's remove all uses of `re` in this PR, `os.path` is much more idiomatic

Reply:
As explained elsewhere, there's no doc on what separator the archives use for filenames within them, and os.path/os.sep might not match.